block-node/block-providers/files.historic/src/main/java/org/hiero/block/node/blocks/files/historic/BlocksFilesHistoricPlugin.java

```diff
@@ -234,17 +260,19 @@ private void attemptZipping() {
     private void startMovingBatchOfBlocksToZipFile(final LongRange batchRange) {
         // check if the batch of blocks is already in progress
         if (inProgressZipRanges.contains(batchRange)) {
+            // if the batch is in progress, we must not submit a task
             LOGGER.log(
                     System.Logger.Level.DEBUG,
                     "Batch of blocks[{%1} -> {%2}] is already in progress",
                     batchRange.start(),
                     batchRange.end());
-            return;
+        } else {
+            // if the batch is not in progress, we must submit a task
+            // add the batch of blocks to the in progress ranges
+            inProgressZipRanges.add(batchRange);
+            // move the batch of blocks to a zip file (submit a task)
+            zipMoveExecutorService.submit(() -> moveBatchOfBlocksToZipFile(batchRange));
```

Comment:
Rather than a lambda, this could be a static inner class.
```suggestion
            zipMoveExecutorService.submit(new BatchToZipFileMover(batchRange));
```
It's a small difference, but it makes maintenance easier and reduces the probability of unplanned threading issues.